consensus/info.go

```diff
@@ -151,15 +179,7 @@ func (s *State) BlocksSince(id BlockID) (removedBlocks, addedBlocks []BlockID, e
 	}
 
 	// Get all the IDs going forward from the common parent.
-	for height := path[0].height + 1; ; height++ {
-		if _, exists := s.currentPath[height]; !exists {
-			break
-		}
-
-		node := s.blockMap[s.currentPath[height]]
-		addedBlocks = append(addedBlocks, node.block.ID())
-	}
-
+	addedBlocks = s.currentPath[path[0].height+1:]
```

Comment:
this code was pretty funny anyway... using the `BlockID` to lookup the `*blockNode` to get the `Block` to calculate the... `BlockID`.